source/funkin/ui/freeplay/FreeplayState.hx

```diff
@@ -523,7 +523,17 @@ class FreeplayState extends MusicBeatSubState
       if (grpCapsules.members.length > 0)
       {
         FunkinSound.playOnce(Paths.sound('scrollMenu'), 0.4);
-        curSelected = 1;
+
+        // This makes it so it sends you to the same song in the new list (if it's there).
+        if (grpCapsules.members[curSelected].freeplayData != null)
+        {
+          curSelected = grpCapsules.members.indexOf(grpCapsules.members[curSelected]);
+        }
+        else
+        {
+          curSelected = 1;
```

Comment:
Personally I would've put `curSelected = 0;` here so that if the list doesn't contain any songs, it takes you to the Random button instead

Great PR nonetheless!